test/integration/scheduler/scheduler_test.go

```diff
@@ -862,4 +887,82 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			framework.ExpectPendingWorkloadsMetric(strictFIFOClusterQ, 2)
 		})
 	})
+
+	ginkgo.When("Deleting clusterQueues", func() {
+		var (
+			cq    *kueue.ClusterQueue
+			queue *kueue.Queue
+		)
+
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("cluster-queue").Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+			queue = testing.MakeQueue("foo-queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			framework.ExpectedClusterQueueToBeDeleted(ctx, k8sClient, cq, false)
+		})
+
+		ginkgo.It("Should not admit new created workloads", func() {
+			ginkgo.By("New created workloads should be admitted")
+			wl1 := testing.MakeWorkload("workload1", ns.Name).Queue(queue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
+			defer func() {
+				gomega.Expect(framework.DeleteWorkload(ctx, k8sClient, wl1)).To(gomega.Succeed())
+			}()
+			framework.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, cq.Name, wl1)
+
+			ginkgo.By("Delete clusterQueue")
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, cq)).To(gomega.Succeed())
+			gomega.Eventually(func() []string {
+				var newCQ kueue.ClusterQueue
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(cq), &newCQ)).To(gomega.Succeed())
+				return newCQ.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(gomega.Equal([]string{kueue.ResourceInUseFinalizerName}))
+
+			ginkgo.By("New created workloads should be frozen")
+			wl2 := testing.MakeWorkload("workload2", ns.Name).Queue(queue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+			defer func() {
+				gomega.Expect(framework.DeleteWorkload(ctx, k8sClient, wl2)).To(gomega.Succeed())
+			}()
+			framework.ExpectWorkloadsToBeFrozen(ctx, k8sClient, cq.Name, wl2)
+		})
+
+		ginkgo.It("Should not become active after creating resourceFlavors", func() {
```

Comment:
I think this test is not correct, the resourceFlavor created below is not referenced by the CQ, so it is not relevant.

What we need to test is the following:
1) Create a CQ that references a non-existent flavor
2) Create a workload, it should not schedule because the flavor doesn't exist
3) Delete the CQ
4) Create the flavor
5) the workload should continue to be frozen

Reply:
hmm, I think we actually need more than that because when the CQ is deleted, it will actually be removed because it has no dependencies, it will not stay in terminating. So this test is not really testing something substantial as is, to make it more valuable we need to do the following:

1. Create CQ that references an existent flavor
2. create a workload that schedules
3. Update the CQ so that it refers to a non-existent flavor
4. create a second workload, it should not schedule because the CQ is in pending
5. delete the CQ, the CQ should continue to exist because the first workload is still running
6. create the flavor
7. the second workload should continue to be frozen because the CQ will be in terminating state

But, I would remove this test completely instead of doing the above, I think the case above is already covered in unit tests.

Reply:
Removed then.